concrete/controllers/single_page/dashboard/system/environment/entities.php

```diff
@@ -2,49 +2,77 @@
 namespace Concrete\Controller\SinglePage\Dashboard\System\Environment;
 
 use Concrete\Core\Database\DatabaseStructureManager;
-use Concrete\Core\Package\Package;
+use Concrete\Core\Package\CustomEntityManagersInterface;
+use Concrete\Core\Package\PackageService;
 use Concrete\Core\Page\Controller\DashboardPageController;
-use Core;
-use Config;
-use Doctrine\ORM\Tools\SchemaTool;
-use ORM;
+use Doctrine\ORM\EntityManagerInterface;
 
 class Entities extends DashboardPageController
 {
     public function view()
     {
         // Retrieve all entity manager drivers and show data about them
-        $config = $this->getEntityManager()->getConfiguration();
-        $driverChain = $config->getMetadataDriverImpl();
-        $drivers = $driverChain->getDrivers();
+        $drivers = [];
+        foreach ($this->getAllEntityManagers() as $em) {
+            $config = $em->getConfiguration();
+            $thisDriverChain = $config->getMetadataDriverImpl();
+            if ($thisDriverChain !== null) {
+                $theseDrivers = $thisDriverChain->getDrivers();
+                if (!empty($theseDrivers)) {
+                    $drivers = array_merge($drivers, $theseDrivers);
+                }
+            }
+        }
+        $this->set('doctrine_dev_mode', (bool) $this->app->make('config')->get('concrete.cache.doctrine_dev_mode'));
         $this->set('drivers', $drivers);
     }
 
-
     public function update_entity_settings()
     {
-        if (!$this->token->validate("update_entity_settings")) {
+        if (!$this->token->validate('update_entity_settings')) {
             $this->error->add($this->token->getErrorMessage());
         }
         if (!$this->error->has()) {
             if ($this->isPost()) {
-                $ddm = $this->post('DOCTRINE_DEV_MODE') == 1 ? 1 : 0;
+                $ddm = $this->post('DOCTRINE_DEV_MODE') === 'yes';
```

Comment:
For these request values it'd be cool to start using `filter_var($variable, FILTER_VALIDATE_BOOLEAN)` to catch things like `"yes", "true", "y", true, 1`

Reply:
Well, the value of the `DOCTRINE_DEV_MODE` checkbox field is not variable: it's always `'yes'` if checked...